src/models/tiles/geometry/geometry-content.test.ts

```diff
@@ -339,31 +353,29 @@ describe("GeometryContent", () => {
 
   it("can add/remove/update polygons", () => {
     const { content, board } = createContentAndBoard();
-    content.addPoints(board, [[1, 1], [3, 3], [5, 1]], [{ id: "p1" }, { id: "p2" }, { id: "p3" }]);
-    expect(content.lastObject).toEqual({ id: "p3", type: "point", x: 5, y: 1 });
-    let polygon: JXG.Polygon | undefined = content.createPolygonFromFreePoints(board) as JXG.Polygon;
-    expect(content.lastObject).toEqual({ id: polygon.id, type: "polygon", points: ["p1", "p2", "p3"] });
+    const { polygon, points } = buildPolygon(board, content, [[1, 1], [3, 3], [5, 1]]);
+    expect(content.lastObjectOfType("polygon")).toEqual({
+      id: polygon?.id, type: "polygon", points: [ points[0].id, points[1].id, points[2].id ] });
     expect(isPolygon(polygon)).toBe(true);
-    const polygonId = polygon.id;
-    expect(polygonId.startsWith("testid-")).toBe(true);
-    expect(content.getDependents(["p1"])).toEqual(["p1", polygonId]);
-    expect(content.getDependents(["p1"], { required: true })).toEqual(["p1"]);
-    expect(content.getDependents(["p3"])).toEqual(["p3", polygonId]);
-    expect(content.getDependents(["p3"], { required: true })).toEqual(["p3"]);
+    const polygonId = polygon?.id;
```

Comment:
FYI, another approach here would be to assert that it's defined:
```suggestion
    const polygonId = polygon!.id;
```
And then you wouldn't have to test for it below. Not something I'd recommend in production code but I think it's fine in tests.